### PR TITLE
Improved formatting of comments and suppressions

### LINKED ISSUE #16681: Documentation comment is indented on "Suppress ... in Source"
**Version Used**: Visual Studio 2017 RC 15.0.26020.0

**Code to Reproduce**:
```csharp
    /// <summary>
    /// 
    /// </summary>
    /// <returns></returns>
    public static async Task<object> Foo()
    {
        return await Task.FromResult(default(object));
    }
```

**Actual Behavior:**
```csharp
#pragma warning disable IDE1006 // Naming Styles
                               /// <summary>
                               /// 
                               /// </summary>
                               /// <returns></returns>
    public static async Task<object> Foo()
#pragma warning restore IDE1006 // Naming Styles
    {
        return await Task.FromResult(default(object));
    }
```

### LINKED ISSUE #38224: IDE0055 incorrectly indents with intermixed comment styles
**Version Used**: 
3.3.0-beta3-19415-01+f5ba9f2c61a2fe853dc4913888d40df221539147

**Steps to Reproduce**:
```C#
using System;
using System.Runtime.InteropServices;

class Program
{
    static void Main() { }

    [DllImport("kernel32.dll")]
    internal static extern bool CreateDirectory(
        /*LPCSTR*/ string path, // this is a nice comment
        /*LPSECURITY_ATTRIBUTES*/ IntPtr lpSecurityAttributes);
}
```

**Expected Behavior**:
IDE0055 not offerred.

**Actual Behavior**:
IDE0055 offered and formats the code to this:
```C#
using System;
using System.Runtime.InteropServices;

class Program
{
    static void Main() { }

    [DllImport("kernel32.dll")]
    internal static extern bool CreateDirectory(
        /*LPCSTR*/ string path, // this is a nice comment
                                /*LPSECURITY_ATTRIBUTES*/ IntPtr lpSecurityAttributes);
}
```

### PR DESCRIPTION
**Pull request by sharwell**
Review by commit is highly recommended to understand the state of the work (each commit is a distinct and non-breaking change).

Fixes #16681 
Fixes #38224

### CODE REVIEW

File: src/Features/CSharp/Portable/CodeFixes/Suppression/CSharpSuppressionCodeFixProvider.cs [hunk 1]
```diff
@@ -79,7 +79,7 @@ protected override bool IsAttributeListWithAssemblyAttributes(SyntaxNode node)
         }
 
         protected override bool IsEndOfLine(SyntaxTrivia trivia)
-            => trivia.Kind() == SyntaxKind.EndOfLineTrivia;
+            => trivia.IsKind(SyntaxKind.EndOfLineTrivia) || trivia.IsKind(SyntaxKind.SingleLineDocumentationCommentTrivia);
```
Comment (mavasani) [inline src/Features/CSharp/Portable/CodeFixes/Suppression/CSharpSuppressionCodeFixProvider.cs]: Please make the same change in VB override and add unit tests for VB as well.

**Comment by sharwell** [comment]
> Please fix the VB suppression fix provider as well.

I'm not aware of bugs there. Do you know of any?

**Comment by mavasani** [comment]
> I'm not aware of bugs there. Do you know of any?

VB has the exact same implementation as C#: http://source.roslyn.io/#Microsoft.CodeAnalysis.VisualBasic.Workspaces/Extensions/SyntaxTriviaExtensions.vb,5ee760c08a07ff52. I suspect it will have the same issue as C# for the unit tests that you added in your last commit. Can you try porting those tests to VB and see the current behavior?

**Comment by sharwell** [comment]
@mavasani I'll leave the VB side for a follow-up. I haven't hit any problems there myself, and haven't heard of any problems from users, so I'll leave it until such time as someone has a need for the change.

File: src/Features/CSharp/Portable/CodeFixes/Suppression/CSharpSuppressionCodeFixProvider.cs (hunk 1 above)
Comment (sharwell) [reply src/Features/CSharp/Portable/CodeFixes/Suppression/CSharpSuppressionCodeFixProvider.cs]: ➡️ I'll leave the VB side for a follow-up. I haven't hit any problems there myself, and haven't heard of any problems from users, so I'll leave it until such time as someone has a need for the change.

**Comment by mavasani** [comment]
Thanks, sounds reasonable to me.

**Pull request merged**